lib/flow/VaporField.cpp

```diff
@@ -237,10 +237,15 @@ int VaporField::GetVelocity(double time, const glm::vec3 &pos, glm::vec3 &veloci
             missingV[i] = grid->GetMissingValue();
         }
         auto  hasMissing = glm::equal(velocity, missingV);
+        // If missing values are represented using NaN, you cannot compare equality with them!
```

Comment:
One small nit: missing values should NEVER be represented as NaNs because, as you say, you can't compare them. The NaN's must be either either in the existing data or generated by the interpolation somehow.

Reply:
I think it's from the specific data set in this case. I.e., `grid->GetMissingValue()` would return you a nan... Does this header look like it's using nan? 

```
netcdf test_bx_vapor {
dimensions:
	time = 1 ;
	x = 168 ;
	y = 93 ;
	z = 46 ;
variables:
	double Bx(time, x, y, z) ;
		Bx:_FillValue = NaN ;
		Bx:units = "G" ;
	double time(time) ;
		time:_FillValue = NaN ;
		time:axis = "T" ;
		time:units = "seconds since 2000-0101" ;
	double x(x) ;
		x:_FillValue = NaN ;
		x:axis = "Z" ;
		x:units = "m" ;
	double y(y) ;
		y:_FillValue = NaN ;
		y:axis = "Y" ;
		y:units = "m" ;
	double z(z) ;
		z:_FillValue = NaN ;
		z:axis = "X" ;
		z:units = "m" ;
}
```

Reply:
You are correct. I thought using NaNs for missing values were prohibited by the CF conventions, but that is not the case. Unfortunately, this means that there are several other places in the code that should be changed to handle missing values that are NaNs. I will open an issue.